altair/vegalite/v5/api.py

```diff
@@ -345,11 +384,641 @@ def check_fields_and_encodings(parameter: Parameter, field_name: str) -> bool:
+        self,
+        statement: _StatementOrLiteralType,
+        *,
+        str_as_lit: bool = ...,
+        seq_as_lit: Literal[True] = ...,
+        **kwargs: Any,
+    ) -> _Conditions: ...
+    @overload
+    def otherwise(
+        self,
+        statement: core.SchemaBase,
+        *,
+        str_as_lit: bool,
+        seq_as_lit: Literal[False],
+        **kwargs: Any,
+    ) -> core.SchemaBase: ...
+    @overload
+    def otherwise(
+        self,
+        statement: _DictOrStr,
+        *,
+        str_as_lit: bool,
+        seq_as_lit: bool,
+        **kwargs: Any,
+    ) -> _Conditions: ...
+    def otherwise(
+        self,
+        statement: _StatementOrLiteralType,
+        *,
+        str_as_lit: bool = False,
+        seq_as_lit: bool = True,
+        **kwargs: Any,
+    ) -> Union[core.SchemaBase, _Conditions]:
+        lit = _LiteralConfig(str_as_lit, seq_as_lit)
+        return _parse_otherwise(statement, self.to_dict(), kwargs, lit)
+
+    def when(
+        self,
+        predicate: _AltOptional[_PredicateType] = Undefined,
+        *more_predicates: _ComposablePredicateType,
+        empty: _AltOptional[bool] = Undefined,
+        **constraints: _FieldEqualType,
+    ) -> "_ChainedWhen":
+        condition = _parse_when(predicate, *more_predicates, empty=empty, **constraints)
+        return _ChainedWhen(condition, self.to_dict())
+
+    def to_dict(self, *args: Any, **kwargs: Any) -> _Conditions:
+        return self._conditions.copy()
+
+
+class _ChainedWhen(_BaseWhen):
+    """
+    Utility class for `when-then-otherwise` conditions.
+
+    Represents the state after calling `alt.when(...).then(...).when(...)`.
+
+    This partial state requires calling `then` to finish the condition.
+
+    References
+    ----------
+    [polars.expr.whenthen](https://github.com/pola-rs/polars/blob/b85c5e0502ca99c77742ee25ba177e6cd11cf100/py-polars/polars/expr/whenthen.py)
+    """
+
+    def __init__(self, condition: _ConditionType, conditions: _Conditions, /) -> None:
+        self._condition = condition
+        self._conditions = conditions
+
+    @overload
+    def then(
+        self,
+        statement: _StatementOrLiteralType,
+        *,
+        str_as_lit: bool = ...,
+        seq_as_lit: Literal[True] = ...,
+        **kwargs: Any,
+    ) -> _Then: ...
+    @overload
+    def then(
+        self,
+        statement: _StatementType,
+        *,
+        str_as_lit: bool,
+        seq_as_lit: Literal[False],
+        **kwargs: Any,
+    ) -> _Then: ...
+    def then(
+        self,
+        statement: _StatementOrLiteralType,
+        *,
+        str_as_lit: bool = False,
+        seq_as_lit: bool = True,
+        **kwargs: Any,
+    ) -> _Then:
+        lit = _LiteralConfig(str_as_lit, seq_as_lit)
+        condition = self._when_then(statement, kwargs, lit)
+        conditions = self._conditions.copy()
+        conditions["condition"].append(condition)
+        return _Then(conditions)
+
+
+def when(
```

Comment:
RFC on `alt.when` docstring.

The passing commit hasn't got the messages, so for context I've added those below. 

Of note, `alt.when`'s doc will be very similar to [`alt._Then.when`](https://github.com/dangotbanned/altair/blob/80a08122ad1e1decdd0986e25a852f928f743e1d/altair/vegalite/v5/api.py#L825) - so I'll be leaving that until after this has been agreed.


`_FieldEqualType` in 989b1f50ee428d4b7de7f591a24731276cbbea1c
> - Provides static type checking instead of `Any` 
> - Alias name hints towards usage
> - Alias doc gives further context, without requiring repeating & syncing wherever it is used

`alt.when` in 7d412198720170a1e4ab46717df931b800e1dee6

> I'm happy with the content in the doc, hopefully this can be refined during review.
> 
> - It adapts parts of `alt.condition`, but not verbatim as there was a `# TODO: update the docstring`
> - Contains links to other sources of inspo
> - Deliberately avoided the word "condition" as much as possible
>   -  `when`'s result can become a condition, but only after calling `result.then(...)`.
>   - Stressing that `when` != `condition`